--- debian/patches/series.org	2023-12-02 19:44:59
+++ debian/patches/series	2023-12-02 19:47:19
@@ -8,30 +8,3 @@
 #buffer-http-request-bodies-for-tlsv13.diff
 #tlsv13-add-logno.diff
 fix-macro.patch
-
-# This patch is applied manually
-#suexec-custom.patch
-CVE-2022-22719.patch
-CVE-2022-22720.patch
-CVE-2022-22721.patch
-CVE-2022-23943-1.patch
-CVE-2022-23943-2.patch
-CVE-2022-26377.patch
-CVE-2022-28614.patch
-CVE-2022-28615.patch
-CVE-2022-29404.patch
-CVE-2022-30522.patch
-CVE-2022-30556.patch
-CVE-2022-31813.patch
-fix-a-possible-listener-deadlock.patch
-handle-children-killed-pathologically.patch
-CVE-2006-20001.patch
-CVE-2022-36760.patch
-CVE-2022-37436.patch
-CVE-2023-25690-1.patch
-CVE-2023-25690-2.patch
-CVE-2023-27522.patch
-mod_proxy_hcheck_jammy_fix_to_detect_support.patch
-reenable-workers-in-standard-error-state-jammy-apache2.patch
-update_http2.patch
-CVE-2023-31122.patch
